File: Laravel/app/Biab/Resources/Blog.php
<?php

namespace App\Biab\Resources;

use App\Biab\Client;

/** Posts, categories, tags, threaded comments, reactions. */
class Blog
{
    public function __construct(private readonly Client $client)
    {
    }

    /** @return array<string, mixed> */
    public function listPosts(?int $limit = null, int|string|null $cursor = null, ?string $categoryId = null): array
    {
        return $this->client->get('blog/posts', [
            'limit' => $limit,
            'cursor' => $cursor,
            'categoryId' => $categoryId,
        ]);
    }

    /** @return array<string, mixed> */
    public function getPost(string $slug): array
    {
        return $this->client->get('blog/posts/'.rawurlencode($slug));
    }

    /** @return array<string, mixed> */
    public function listCategories(): array
    {
        return $this->client->get('blog/categories');
    }

    /** @return array<string, mixed> */
    public function listTags(): array
    {
        return $this->client->get('blog/tags');
    }

    /** @return array<string, mixed> */
    public function listComments(string $slug, ?int $limit = null, int|string|null $cursor = null): array
    {
        return $this->client->get('blog/posts/'.rawurlencode($slug).'/comments', [
            'limit' => $limit,
            'cursor' => $cursor,
        ]);
    }
}

File: Phoenix/lib/bd_starter/bd/resources/blog.ex
defmodule BdStarter.Bd.Resources.Blog do
  @moduledoc "Posts, categories, tags, threaded comments."

  alias BdStarter.Bd.Client

  def list(client, limit \\ 20, cursor \\ nil),
    do: Client.get(client, "blog/posts", limit: limit, cursor: cursor)

  def get(client, slug), do: Client.get(client, "blog/posts/#{enc(slug)}")

  def comments(client, slug, limit \\ 50),
    do: Client.get(client, "blog/posts/#{enc(slug)}/comments", limit: limit)

  def categories(client), do: Client.get(client, "blog/categories")
  def tags(client), do: Client.get(client, "blog/tags")

  defp enc(v), do: URI.encode(to_string(v), &URI.char_unreserved?/1)
end
